### PR TITLE
FINERACT-1932: Add custom Configuration class for transfer service

### PR DESCRIPTION
**Pull request by mmagdiab**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.apache.fineract.portfolio.transfer.service.TransferWritePlatformServiceJpaRepositoryImpl;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class TransferAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java]: Please provide the Java interface class reference in parenthesis with the `@ConditionalOnMissingBean` annotation.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/service/TransferWritePlatformServiceJpaRepositoryImpl.java [hunk 2]
```diff
@@ -83,7 +80,6 @@ public class TransferWritePlatformServiceJpaRepositoryImpl implements TransferWr
     private final ClientTransferDetailsRepositoryWrapper clientTransferDetailsRepositoryWrapper;
     private final PlatformSecurityContext context;
 
-    @Autowired
     public TransferWritePlatformServiceJpaRepositoryImpl(final ClientRepositoryWrapper clientRepositoryWrapper,
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/service/TransferWritePlatformServiceJpaRepositoryImpl.java]: Please use Lombok annotations to get rid of the constructor. Let's clean up when we have the opportunity.

Comment (mmagdiab) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/service/TransferWritePlatformServiceJpaRepositoryImpl.java]: done

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java (hunk 1 above)
Comment (mmagdiab) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java]: done

**Review by adamsaghy — APPROVED** [review]
LGTM

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.apache.fineract.portfolio.savings.service.SavingsAccountWritePlatformService;
+import org.apache.fineract.portfolio.transfer.data.TransfersDataValidator;
+import org.apache.fineract.portfolio.transfer.service.TransferWritePlatformService;
+import org.apache.fineract.portfolio.transfer.service.TransferWritePlatformServiceJpaRepositoryImpl;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class TransferAutoConfiguration {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java]: This is not **Auto**Configuration but a regular Configuration, Please rename it accordingly, just like in your other PRs. Thanks.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java (hunk 3 above)
Comment (mmagdiab) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java]: Thanks @galovics, as this configuration class should be check for availability of another custom "over-riding" configurations. I will change the annotation to `@AutoConfiguration `so it runs after normal `@Configuration` in the lifecycle.

Comment (galovics) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java]: I still see the configuration class as AutoConfiguration.

Comment (mmagdiab) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/transfer/starter/TransferAutoConfiguration.java]: updates to be Configuration class

**Review by galovics — APPROVED** [review]

**Comment by adamsaghy** [comment]
@mmagdiab Please kindly rebase this PR with the latest develop branch. The failing tests were already addressed.

**Comment by mmagdiab** [comment]
> @mmagdiab Please kindly rebase this PR with the latest develop branch. The failing tests were already addressed.

resolved

**Pull request merged**